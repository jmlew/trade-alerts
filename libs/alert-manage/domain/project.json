{
  "$schema": "../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/alert-manage/domain/src",
  "projectType": "library",
  "targets": {
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/alert-manage/domain/**/*.ts"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["coverage/libs/alert-manage/domain"],
      "options": {
        "jestConfig": "libs/alert-manage/domain/jest.config.ts",
        "passWithNoTests": true
      }
    }
  },
  "tags": ["type:domain-logic", "domain:alert-manage"]
}
