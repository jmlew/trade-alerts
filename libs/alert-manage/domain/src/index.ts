// All domain entities exposed to other libs.
export * from './lib/entities/alert-manager-api.enum';
export * from './lib/entities/alert-manager-data.enum';
export * from './lib/entities/alert-manager-data.model';

// Expose only the facades which encapsulates all data services.
export * from './lib/application/alert-manager.facade';
