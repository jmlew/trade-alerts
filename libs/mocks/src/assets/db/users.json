{
  "data": [
    {
      "id": 1,
      "firstName": "John",
      "lastName": "Smith",
      "email": "john@sampleuser.com",
      "createdAt": "2018-07-16T07:32:20.421Z"
    },
    {
      "id": 2,
      "firstName": "Fred",
      "lastName": "Jackson",
      "email": "fred@sampleuser.com",
      "createdAt": "2018-07-16T07:32:20.421Z"
    },
    {
      "id": 3,
      "firstName": "Sam",
      "lastName": "Harris",
      "email": "sam@sampleuser.com",
      "createdAt": "2018-07-18T07:32:20.421Z"
    }
  ]
}
