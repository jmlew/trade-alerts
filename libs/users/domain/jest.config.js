module.exports = {
  displayName: 'users-domain',
  preset: '../../../jest.preset.js',
  transform: {
    '^.+\\.[tj]sx?$': 'babel-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../../coverage/libs/users/domain',
};
