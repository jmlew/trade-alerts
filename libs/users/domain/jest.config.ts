/* eslint-disable */
export default {
  displayName: 'users-domain',
  preset: '../../../jest.preset.js',
  transform: {
    '^.+\\.[tj]sx?$': ['babel-jest', { presets: ['@nrwl/react/babel'] }],
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../../coverage/libs/users/domain',
};
