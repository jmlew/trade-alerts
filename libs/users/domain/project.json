{
  "$schema": "../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/users/domain/src",
  "projectType": "library",
  "tags": ["domain:users", "type:domain-logic"],
  "targets": {
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/users/domain/**/*.{ts,tsx,js,jsx}"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["coverage/libs/users/domain"],
      "options": {
        "jestConfig": "libs/users/domain/jest.config.ts",
        "passWithNoTests": true
      }
    }
  }
}
