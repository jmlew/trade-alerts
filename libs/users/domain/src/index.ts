export * from './lib/application/user.facade';
export * from './lib/entities/user.model';
export * from './lib/infrastructure/user-api.model';
