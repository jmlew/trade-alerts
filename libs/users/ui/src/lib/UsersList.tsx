import { Delete, Edit } from '@mui/icons-material';
import {
  Avatar,
  IconButton,
  List,
  ListItem,
  ListItemAvatar,
  ListItemText,
} from '@mui/material';
import { UserRecord } from '@trade-alerts/users/domain';

export interface UsersListProps {
  users: UserRecord[];
  isControlsHidden?: boolean;
  onEditUser: (userId: number) => void;
  onDeleteUser: (userId: number) => void;
}

export function UsersList({
  users,
  isControlsHidden,
  onEditUser,
  onDeleteUser,
}: UsersListProps) {
  return (
    <List>
      {users.map((user: UserRecord) => (
        <UsersListItem
          key={user.id}
          user={user}
          isControlsHidden={isControlsHidden ?? false}
          onEdit={() => onEditUser(user.id)}
          onDelete={() => onDeleteUser(user.id)}
        />
      ))}
    </List>
  );
}

interface UserItemProps {
  user: UserRecord;
  isControlsHidden: boolean;
  onEdit: () => void;
  onDelete: () => void;
}

function UsersListItem({ user, isControlsHidden, onEdit, onDelete }: UserItemProps) {
  return (
    <ListItem
      divider={false}
      secondaryAction={
        isControlsHidden ? null : (
          <>
            <IconButton edge="start" aria-label="edit" onClick={onEdit}>
              <Edit />
            </IconButton>
            <IconButton edge="end" aria-label="edit" onClick={onDelete}>
              <Delete />
            </IconButton>
          </>
        )
      }
    >
      <ListItemAvatar>
        <Avatar src={user.avatar} />
      </ListItemAvatar>
      <ListItemText primary={user.firstName} secondary={user.lastName || null} />
    </ListItem>
  );
}
