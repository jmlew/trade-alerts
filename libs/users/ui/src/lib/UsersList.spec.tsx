import renderer from 'react-test-renderer';

import { fireEvent, render } from '@testing-library/react';
import { User } from '@trade-alerts/users/domain';

import { UsersList } from './UsersList';

function setup() {
  function onEditUser(userId: number) {
    // default
  }
  function onDeleteUser(userId: number) {
    // default
  }
  const users: User[] = [
    { id: 1, firstName: 'Jason', lastName: 'Lewis', email: 'jason@localhost.com' },
    { id: 2, firstName: 'System', lastName: 'Infra', email: 'system@localhost.com' },
    { id: 3, firstName: 'Administrator', lastName: 'Main', email: 'admin@localhost.com' },
  ];
  return { users, onEditUser, onDeleteUser };
}

describe(UsersList, () => {
  it('should render the component correctly', () => {
    const { users, onEditUser, onDeleteUser } = setup();
    const component = renderer.create(
      <UsersList users={users} onEditUser={onEditUser} onDeleteUser={onDeleteUser} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render a list item for each user', () => {
    const { users, onEditUser, onDeleteUser } = setup();
    const { getByTestId } = render(
      <UsersList users={users} onEditUser={onEditUser} onDeleteUser={onDeleteUser} />
    );
    const length = getByTestId('list').children.length;
    expect(length).toEqual(users.length);
  });
  it('should call edit on edit button click', () => {
    const { users, onDeleteUser } = setup();
    let currentUserId = null;
    function onEditUser(userId: number) {
      currentUserId = userId;
    }
    const { getAllByLabelText } = render(
      <UsersList users={users} onEditUser={onEditUser} onDeleteUser={onDeleteUser} />
    );
    const index = 1;
    const user = users[index];
    const btn = getAllByLabelText('edit')[index];
    expect(btn).toBeDefined();
    btn && fireEvent.click(btn);
    expect(currentUserId).toBe(user.id);
  });
  it('should call delete on delete button click', () => {
    const { users, onEditUser } = setup();
    let currentUserId = null;
    function onDeleteUser(userId: number) {
      currentUserId = userId;
    }
    const { getAllByLabelText } = render(
      <UsersList users={users} onEditUser={onEditUser} onDeleteUser={onDeleteUser} />
    );
    const index = 1;
    const user = users[index];
    const btn = getAllByLabelText('delete')[index];
    expect(btn).toBeDefined();
    btn && fireEvent.click(btn);
    expect(currentUserId).toBe(user.id);
  });
});
