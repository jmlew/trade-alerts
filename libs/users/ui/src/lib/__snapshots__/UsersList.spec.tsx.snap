// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersList should render the component correctly 1`] = `
<ul
  className="MuiList-root MuiList-padding css-h4y409-MuiList-root"
  data-testid="list"
>
  <li
    className="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1rki8sg-MuiListItem-root"
    data-cy-list-item={1}
    disabled={false}
  >
    <div
      className="MuiListItemAvatar-root css-1e9lk82-MuiListItemAvatar-root"
    >
      <div
        className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
          data-testid="PersonIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
          />
        </svg>
      </div>
    </div>
    <div
      className="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
      >
        Jason
      </span>
      <p
        className="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-83ijpv-MuiTypography-root"
      >
        Lewis
      </p>
    </div>
    <div
      className="MuiListItemSecondaryAction-root css-518kzi-MuiListItemSecondaryAction-root"
    >
      <button
        aria-label="edit"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeStart MuiIconButton-sizeMedium css-9sors8-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="EditIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
          />
        </svg>
      </button>
      <button
        aria-label="delete"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="DeleteIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
          />
        </svg>
      </button>
    </div>
  </li>
  <li
    className="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1rki8sg-MuiListItem-root"
    data-cy-list-item={2}
    disabled={false}
  >
    <div
      className="MuiListItemAvatar-root css-1e9lk82-MuiListItemAvatar-root"
    >
      <div
        className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
          data-testid="PersonIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
          />
        </svg>
      </div>
    </div>
    <div
      className="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
      >
        System
      </span>
      <p
        className="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-83ijpv-MuiTypography-root"
      >
        Infra
      </p>
    </div>
    <div
      className="MuiListItemSecondaryAction-root css-518kzi-MuiListItemSecondaryAction-root"
    >
      <button
        aria-label="edit"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeStart MuiIconButton-sizeMedium css-9sors8-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="EditIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
          />
        </svg>
      </button>
      <button
        aria-label="delete"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="DeleteIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
          />
        </svg>
      </button>
    </div>
  </li>
  <li
    className="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1rki8sg-MuiListItem-root"
    data-cy-list-item={3}
    disabled={false}
  >
    <div
      className="MuiListItemAvatar-root css-1e9lk82-MuiListItemAvatar-root"
    >
      <div
        className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
          data-testid="PersonIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
          />
        </svg>
      </div>
    </div>
    <div
      className="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
      >
        Administrator
      </span>
      <p
        className="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-83ijpv-MuiTypography-root"
      >
        Main
      </p>
    </div>
    <div
      className="MuiListItemSecondaryAction-root css-518kzi-MuiListItemSecondaryAction-root"
    >
      <button
        aria-label="edit"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeStart MuiIconButton-sizeMedium css-9sors8-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="EditIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
          />
        </svg>
      </button>
      <button
        aria-label="delete"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="DeleteIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
          />
        </svg>
      </button>
    </div>
  </li>
</ul>
`;
